client/my-sites/email/email-management/home/utils.ts

```diff
@@ -219,12 +220,21 @@ export function recordInboxNewMailboxUpsellClickEvent() {
 }
 
 /**
- * Tracks an event for the key 'calypso_inbox_upsell'.
+ * Tracks an event for the key 'calypso_{source}_upsell', where {source} defaults to "email".
+ * For upsell triggered from the inbox, the event 'calypso_inbox_upsell' will be tracked.
  *
+ * @param source - source generating the event.
  * @param context context, where this event was logged.
  */
-export function recordInboxUpsellTracksEvent( context: string | null = null ) {
-	recordTracksEvent( 'calypso_inbox_upsell', {
+export function recordEmailUpsellTracksEvent(
+	source: string | null = null,
+	context: string | null = null
+) {
+	if ( ! source ) {
+		source = 'email';
+	}
+
+	recordTracksEvent( 'calypso_' + source + '_upsell', {
```

Comment:
Although string concatenation is faster than interpolation, please, use string interpolation as is easier to read and is the common practice 🙂 
```suggestion
	recordTracksEvent( `calypso_${ source }_upsell`, {
```